pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -63,54 +63,47 @@ func (w *HashAggFinalWorker) getPartialInput() (input *HashAggIntermData, ok boo
 }
 
 func (w *HashAggFinalWorker) consumeIntermData(sctx sessionctx.Context) (err error) {
-	var (
-		input            *HashAggIntermData
-		ok               bool
-		intermDataBuffer [][]aggfuncs.PartialResult
-		groupKeys        []string
-		sc               = sctx.GetSessionVars().StmtCtx
-	)
 	for {
 		waitStart := time.Now()
-		input, ok = w.getPartialInput()
+		input, ok := w.getPartialInput()
 		if w.stats != nil {
 			w.stats.WaitTime += int64(time.Since(waitStart))
 		}
 		if !ok {
 			return nil
 		}
-		execStart := time.Now()
-		if intermDataBuffer == nil {
-			intermDataBuffer = make([][]aggfuncs.PartialResult, 0, w.maxChunkSize)
+
+		// As the w.partialResultMap is empty when we get the first input.
+		// So it's better to directly assign the input to w.partialResultMap
+		if w.isFirstInput {
+			w.isFirstInput = false
+			w.partialResultMap = *input
+			continue
 		}
-		// Consume input in batches, size of every batch is less than w.maxChunkSize.
-		for reachEnd := false; !reachEnd; {
-			intermDataBuffer, groupKeys, reachEnd = input.getPartialResultBatch(sc, intermDataBuffer[:0], w.aggFuncs, w.maxChunkSize)
-			groupKeysLen := len(groupKeys)
-			memSize := getGroupKeyMemUsage(w.groupKeys)
-			w.groupKeys = w.groupKeys[:0]
-			for i := 0; i < groupKeysLen; i++ {
-				w.groupKeys = append(w.groupKeys, []byte(groupKeys[i]))
+
+		failpoint.Inject("ConsumeRandomPanic", nil)
+
+		execStart := time.Now()
+		allMemDelta := int64(0)
+		for key, value := range *input {
+			dstVal, ok := w.partialResultMap[key]
+			if !ok {
+				// The input will be released, so it's equal to transfer the memory of
+				// input to w.partialResultMap and needless to track memory here.
+				w.partialResultMap[key] = value
```

Comment:
In extreme cases, when all keys from other maps have never appeared in this map, the length of this map will keep increasing. 
So, should we consider the memory consumption resulting from the growth of the underlying data structure of the `w.partialResultMap`?

Reply:
> In extreme cases, when all keys from other maps have never appeared in this map, the length of this map will keep increasing. So, should we consider the memory consumption resulting from the growth of the underlying data structure of the `w.partialResultMap`?

For the stability of tidb, I think conservative strategy is better and I will increase the memory consumption of these maps.